src/libsyntax/parse/parser.rs

```diff
@@ -1353,13 +1352,15 @@ impl<'a> Parser<'a> {
             self.expect(&token::CloseDelim(token::Bracket))?;
             t
         } else if self.check(&token::BinOp(token::And)) ||
-                  self.token == token::AndAnd {
+                  self.check(&token::AndAnd) {
             // BORROWED POINTER
             self.expect_and()?;
             self.parse_borrowed_pointee()?
         } else if self.check_keyword(keywords::For) {
+            // FIXME plus priority
```

Comment:
`+` still has high priority in some cases.

The first case is when trait object type starts with `for`:
```
trait Tr<'a> {}
fn f(arg: &for<'a> Tr<'a> + Send){} // Parsed as &(for<'a> Tr<'a> + Send)
```
I'm going to fix this in upcoming PR so I didn't create an issue.

The second case is `impl Trait1 + Trait2` and it seems to be [known](https://github.com/rust-lang/rust/issues/34511#issue-162545419).